arelle/ValidateXbrlCalcs.py

```diff
@@ -278,6 +338,43 @@ def bindFacts(self, facts, ancestors):
                 if concept in self.conceptsInRequiresElement:
                     self.requiresElementFacts[concept].append(f)
 
+    def oimConsistentInterval(self, fList):
+        if any(f.isNil for f in fList):
+            if all(f.isNil for f in fList):
+                return (NIL_FACT_SET,NIL_FACT_SET)
+            _inConsistent = True # provide error message 
+        else: # not all have same decimals
+            f0 = fList[0]
+            _d = inferredDecimals(f0)
+            _v = f0.xValue
+            _inConsistent = isnan(_v) # NaN is incomparable, always makes dups inconsistent
+            decVals = {_d: _v}
+            aMax, bMin = rangeValue(_v, _d)
+            for f in fList[1:]:
+                _d = inferredDecimals(f)
+                _v = f.xValue
+                if isnan(_v):
+                    _inConsistent = True
+                    break
+                if _d in decVals:
+                    _inConsistent |= _v != decVals[_d]
+                else:
+                    decVals[_d] = _v
+                a, b = rangeValue(_v, _d)
+                if a > aMax: aMax = a
+                if b < bMin: bMin = b
```

Comment:
This is fine, but it suggest moving these lines into the `else`?  Either this is the first time we've seen this value at this decimal (and we need to expand the range), or we're just updating `_inConsistent`.

Reply:
I understand, so in the case where we have multiple values at the same decimals, either the secondary values equal the first (so `aMax`/`bMin` won't change) or `_inConsistent will be set, so we could just write:
```
                if _d in decVals:
                    _inConsistent |= _v != decVals[_d]
                else:
                    decVals[_d] = _v
                    a, b = rangeValue(_v, _d)
                    if a > aMax: aMax = a
                    if b < bMin: bMin = b
```

Reply:
done